core/controllers/email_dashboard_test.py

```diff
@@ -47,46 +47,49 @@ def setUp(self):
         self.set_admins([self.SUBMITTER_USERNAME])
 
     def test_that_handler_works_correctly(self):
-        self.login(self.SUBMITTER_EMAIL)
-        csrf_token = self.get_new_csrf_token()
-        self.post_json(
-            '/emaildashboarddatahandler', {
-                'data': {
-                    'has_not_logged_in_for_n_days': 2,
-                    'inactive_in_last_n_days': 5,
-                    'created_at_least_n_exps': 1,
-                    'created_fewer_than_n_exps': None,
-                    'edited_at_least_n_exps': None,
-                    'edited_fewer_than_n_exps': 2
-                }}, csrf_token=csrf_token)
-        self.logout()
+        swap_api = self.swap(feconf, 'MAILGUN_API_KEY', 'key')
+        swap_domain = self.swap(feconf, 'MAILGUN_DOMAIN_NAME', 'domain')
+        with swap_api, swap_domain:
```

Comment:
Have the part within the "with" be as small as possible. I'm pretty certain you don't need it for everything that's here.